src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -186,7 +186,7 @@ internal static class PeWriter
                 debugDirectoryBuilder = null;
             }
 
-            var peBuilder = new ManagedPEBuilder(
+            var peBuilder = new ExtendedPEBuilder(
```

Comment:
Tried that approach, but it made logic below more complicated (`ExtendedPeBuilder` has an overload of `Serialize` with an extra parameter).